rtx/README.md

```diff
@@ -4,3 +4,10 @@ Contains a function `Must` that calls `log.Fatal` with a nice error message if
 its first argument is not a nil error.  Using `Must` aids in the pursuit of 100%
 code coverage, because it means that the error pathway of `log.Fatal` is in this
 package instead of inline with the code being tested.
+
+A similar function `Should` is provided, with the difference that the program
+won't exit after the error message is printed, and it's meant to be used for
+non-critical errors. A possible usage is to simplify deferred calls to close
+resources:
+
+`defer Should(resources.Close(), "Helpful message")`
```

Comment:
This code, as written, doesn't work.

`defer` does not call the outermost function, but it does call all the functions required to make its arguments. So this code will immediately call `resources.Close()`, and then wait until the block exits to call `Should` with the return value of `resources.Close()`.

What you really want is something like:

```go
func LogOnError(f func() error) func() {
  return func() {
    if err := f(); err != nil {
      log.Println(err)
    }
  }
}
```

Then you could write 

```go
defer LogOnError(resources.Close)  // Note the lack of parentheses on Close
```

But that only works for functions that take no arguments and return an error - which may not be enough of a use case? That version of `LogOnError` would require us to rewrite code like:
```go
defer exitFunc(arg1, arg2)
```
into
```go
defer LogOnError(func() error { return exitFunc(arg1, arg2) })
```
, which doesn't feel like a readability win.